Microsoft.Toolkit.Services/Services/MicrosoftTranslator/ITranslatorService.cs

```diff
@@ -38,35 +38,93 @@ public interface ITranslatorService
+        /// <returns>A <see cref="DetectLanguageResponse"/> object containing information about the detected language.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <list type="bullet">
+        /// <term>The <see cref="SubscriptionKey"/> property hasn't been set.</term>
+        /// <term>The <paramref name="input"/> parameter is <strong>null</strong> (<strong>Nothing</strong> in Visual Basic) or empty.</term>
         /// </list>
         /// </exception>
         /// <exception cref="TranslatorServiceException">The provided <see cref="SubscriptionKey"/> isn't valid or has expired.</exception>
         /// <remarks><para>This method performs a non-blocking request for language detection.</para>
-        /// <para>For more information, go to https://docs.microsofttranslator.com/text-translate.html#!/default/get_Detect.
+        /// <para>For more information, go to https://docs.microsoft.com/azure/cognitive-services/translator/reference/v3-0-detect.
         /// </para></remarks>
         /// <seealso cref="GetLanguagesAsync"/>
         /// <seealso cref="Language"/>
-        Task<string> DetectLanguageAsync(string text);
+        Task<DetectLanguageResponse> DetectLanguageWithResponseAsync(string input);
+
+        /// <summary>
+        /// Detects the language of a text.
+        /// </summary>
+        /// <param name="input">A string array containing the sentences whose language must be detected.</param>
+        /// <returns>A <see cref="DetectLanguageResponse"/> array with one result for each string in the input array. Each object contains information about the detected language.</returns>
+        /// <exception cref="ArgumentException">
+        /// <list type="bullet">
+        /// <term>The <paramref name="input"/> parameter doesn't contain any element.</term>
+        /// <term>The <paramref name="input"/> array contains more than 100 elements.</term>
+        /// </list>
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// <list type="bullet">
+        /// <term>The <see cref="SubscriptionKey"/> property hasn't been set.</term>
+        /// <term>The <paramref name="input"/> array is <strong>null</strong> (<strong>Nothing</strong> in Visual Basic).</term>
+        /// </list>
+        /// </exception>
+        /// <exception cref="TranslatorServiceException">The provided <see cref="SubscriptionKey"/> isn't valid or has expired.</exception>
+        /// <remarks><para>This method performs a non-blocking request for language detection.</para>
+        /// <para>For more information, go to https://docs.microsoft.com/azure/cognitive-services/translator/reference/v3-0-detect.
+        /// </para></remarks>
+        /// <seealso cref="GetLanguagesAsync"/>
+        /// <seealso cref="Language"/>
+        Task<IEnumerable<DetectLanguageResponse>> DetectLanguageWithResponseAsync(IEnumerable<string> input);
```

Comment:
Should this be named plural: `DetectLanugagesWithResponseAsync`?

Reply:
Done.